src/mounts.rs

```diff
@@ -503,23 +518,31 @@ fn readonly_path(path: &str) -> Result<()> {
         Some(&path[1..]),
         path,
         None::<&str>,
-        MS_BIND | MS_REC,
+        MsFlags::MS_BIND | MsFlags::MS_REC,
         None::<&str>,
-    )
-    {
-        // ignore ENOENT: path to make read only doesn't exist
-        if e != Sys(Errno::ENOENT) {
-            let msg = format!("could not readonly {}", path);
-            Err(e).chain_err(|| msg)?;
+    ) {
+        match e {
+            ::nix::Error::Sys(errno) => {
+                // ignore ENOENT: path to make read only doesn't exist
+                if errno != Errno::ENOENT {
+                    let msg = format!("could not readonly {}", path);
+                    Err(e).chain_err(|| msg)?;
+                }
+                debug!("ignoring remount of {} because it doesn't exist", path);
+            }
+            _ => {
```

Comment:
I think that this is actually unreachable. Afaict mount will always return an error of type ::nix::Error::Sys. If we are just using the match to unpack the errno, I think we can switch back to the if statement instead of using a match here.

Reply:
Note that if I'm wrong and this is reachable, I don't think we want to return ok if we hit an unknown error.

Reply:
Alright, that wasn't quite clear to me.